### PR TITLE
Task5 user auto authentication

### PR DESCRIPTION
**Pull request by nitzanpap**

### CODE REVIEW

File: lib/api/firebase/firebase_auth_token_response.dart [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  Map<String, dynamic> toJson() => {
+        "error": error?.toJson(),
+        "kind": kind,
+        "users": users == null
+            ? []
+            : List<dynamic>.from(users!.map((x) => x.toJson())),
+      };
+}
+
+class User {
```
Comment (abenartz) [inline lib/api/firebase/firebase_auth_token_response.dart:46]: Better to separate model classes to different files.
This should be a user under model directory.

File: lib/api/firebase/firebase_auth_token_response.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            ? []
+            : List<dynamic>.from(providerUserInfo!.map((x) => x.toJson())),
+        "validSince": validSince,
+        "lastLoginAt": lastLoginAt,
+        "createdAt": createdAt,
+        "lastRefreshAt": lastRefreshAt?.toIso8601String(),
+      };
+}
+
+class ProviderUserInfo {
```
Comment (abenartz) [inline lib/api/firebase/firebase_auth_token_response.dart:105]: Same.. Extract to different file

File: lib/helpers/helper_functions/device_storage_functions.dart [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+Future<void> saveInPreferencesStorage(
```
Comment (abenartz) [inline lib/helpers/helper_functions/device_storage_functions.dart:3]: Build class called SharedPreferencesManager.
All these classes should be static function inside him.

**Review by abenartz — CHANGES_REQUESTED** [review]

File: lib/api/firebase/firebase_auth_token_response.dart [hunk 4]
```diff
@@ -0,0 +1,132 @@
+// To parse this JSON data, do
+//
+//     final firebaseAuthTokenResponse = firebaseAuthTokenResponseFromJson(jsonString);
+
+import 'dart:convert';
+
+import 'package:dispatcher/api/firebase/firebase_auth_response_error.dart';
+
+FirebaseAuthTokenResponse firebaseAuthTokenResponseFromJson(String str) =>
```
[the remark this answers is not in the document]
Comment (nitzanpap) [reply lib/api/firebase/firebase_auth_token_response.dart:10]: @abenartz Done.

File: lib/api/firebase/firebase_auth_token_response.dart (hunk 1 above)
Comment (nitzanpap) [reply lib/api/firebase/firebase_auth_token_response.dart]: @abenartz Done.

File: lib/api/firebase/firebase_auth_token_response.dart (hunk 2 above)
Comment (nitzanpap) [reply lib/api/firebase/firebase_auth_token_response.dart]: @abenartz Done.

File: lib/helpers/helper_functions/device_storage_functions.dart (hunk 3 above)
Comment (nitzanpap) [reply lib/helpers/helper_functions/device_storage_functions.dart]: @abenartz Done in all relevant places.

**Review by abenartz — APPROVED** [review]

**Pull request merged**